environment/src/main/java/com/sequenceiq/environment/environment/flow/config/update/EnvStackConfigUpdatesActions.java

```diff
@@ -0,0 +1,120 @@
+        this.environmentStatusUpdateService = environmentStatusUpdateService;
+        this.metricService = metricService;
+    }
+
+    @Bean(name = "STACK_CONFIG_UPDATES_START_STATE")
+    public Action<?, ?> collectClusterInfo() {
+        return new AbstractEnvStackConfigUpdatesAction<>(EnvStackConfigUpdatesEvent.class) {
+
+            @Override
+            protected void doExecute(
+                CommonContext context, EnvStackConfigUpdatesEvent payload,
+                Map<Object, Object> variables) {
+
+                EnvironmentDto environmentDto = environmentStatusUpdateService
+                    .updateEnvironmentStatusAndNotify(context, payload,
+                        getCurrentStatus(payload.getResourceId()),
+                        ResourceEvent.ENVIRONMENT_STACK_CONFIGS_UPDATE_STARTED,
+                        EnvStackConfigUpdatesState.STACK_CONFIG_UPDATES_START_STATE);
+
+                sendEvent(context,
+                    EnvStackConfigUpdatesHandlerSelectors.STACK_CONFIG_UPDATES_HANDLER_EVENT
+                        .selector(), environmentDto);
+            }
+        };
+    }
+
+    @Bean(name = "STACK_CONFIG_UPDATES_FINISHED_STATE")
+    public Action<?, ?> finishedAction() {
+        return new AbstractEnvStackConfigUpdatesAction<>(ResourceCrnPayload.class) {
+            @Override
+            protected void doExecute(CommonContext context, ResourceCrnPayload payload,
+                Map<Object, Object> variables) {
+
+                EnvironmentDto environmentDto = environmentStatusUpdateService
+                    .updateEnvironmentStatusAndNotify(context, payload,
+                        getCurrentStatus(payload.getResourceId()),
+                        ResourceEvent.ENVIRONMENT_STACK_CONFIGS_UPDATE_FINISHED,
+                        EnvStackConfigUpdatesState.STACK_CONFIG_UPDATES_FINISHED_STATE);
+                metricService.incrementMetricCounter(MetricType.ENV_STACK_CONFIG_UPDATE_FINISHED,
+                    environmentDto);
+                LOGGER.info("Flow entered into ENV_STOP_FINISHED_STATE");
+                sendEvent(context, FINALIZE_ENV_STACK_CONIFG_UPDATES_EVENT.event(), payload);
+            }
+        };
+    }
+
+    @Bean(name = "STACK_CONFIG_UPDATES_FAILED_STATE")
+    public Action<?, ?> failedAction() {
+        return new AbstractEnvStackConfigUpdatesAction<>(EnvStopFailedEvent.class) {
+            @Override
+            protected void doExecute(CommonContext context, EnvStopFailedEvent payload,
+                Map<Object, Object> variables) {
+                LOGGER.warn(
+                    String.format("Failed to update environments stack configs '%s'. Status: '%s'.",
+                        payload.getEnvironmentDto(), payload.getEnvironmentStatus()),
+                    payload.getException());
+                EnvironmentDto environmentDto = environmentStatusUpdateService
+                    .updateFailedEnvironmentStatusAndNotify(context, payload,
```

Comment:
The environment state itself is being maintained.  
This flow does not really effect the state of the environment it is effecting the 
clusters in it.  The overall envrionment state should not change from what it is
unless its effect by the state of Environment level clustrers in it,  live FreeIPA 
which is not hit by this.

Reply:
then what's the purpose of invoking this method in these steps? only sending notification?
I'm just confused why we need to lookup the env from db, update the status to the same, save it back if it's state won't change. maybe this is the right way to send notification in case of env service, it's just looks there are some extra steps we don't need here.

Reply:
Its to update the ResourceEvent status text that's associated with the Environment.  When the flow is going it updates those in the events view of the Clusters for the config.  This way there is a notification of the event to the customer without adjusting the status of the environment.

Reply:
thanks for the clarification